simplegrid/cow.py

```diff
@@ -116,3 +119,32 @@ def step(self, observation):
             return random.choice(interesting_actions)
         else:
             return Action.RIGHT  # Was: random.choice(possible_actions)
+
+
+class SmartCow(SimpleCow):
+    def step(self, observation):
+
+        if self.energy > MAX_ENERGY:
+            self.actioncolor = YELLOW
+            return Action.SPLIT
+
+        size = observation.shape[0]
+        view_distance = size // 2
+        possible_actions = {a: 0.0 for a in list(Action)[1:-1]}
```

Comment:
Consider using a defaultdict(float)

Reply:
:-) I started it of with a default dict but changed it to a normal dict when the  list(Action)[1:-1] part came along. How would you do that?

Reply:
Why do you need `list(Action)[1:-1]` to begin with? In fact Counter() should do the trick, no? That way you wouldn't need `max(...)` either.